### ISSUE TITLE
ES6 Syntax added in 0.16.2

### ISSUE THREAD

**Issue by mixonic** [ef4/liquid-fire#172]
Howdy! In the 0.16.2 release, it looks like some ES2015 got into the codebase:

https://github.com/ef4/liquid-fire/compare/v0.16.1...v0.16.2#diff-1680ae7a6e04412ce4ec8484851dc5acR14

Is the intent that using a full transpiler is now required for liquid-fire, or should this still be ES5 with ES6 modules? I'd certainly prefer not needing to use a full transpiler.

**Comment by jdhines** [comment]
This causes me to not even be able to `ember s` my app because it hangs up on this point.

**Comment by ef4** [comment]
So I'm sympathetic to this being a hassle for people who don't currently use the full transpiler. And it will not be a big deal to eliminate this little bit of new syntax.

But realize that it's only an implementation detail that ember-cli currently handles module transpilation separate from other ES2015 features. Your app *does* already require ES2015 if your'e using ember-cli and saying `import`. So being picky about which features are ok is not really tenable going forward.

We don't want the whole addon ecosystem to be limited to only the lowest common denominator of ES2015 features forever.

Is your concern increased build times? I'm curious what the impact actually is, because it should be a lot faster now than it was a couple months ago. Enabling is as simple as `npm install --save-dev ember-cli-6to5`

**Comment by mixonic** [comment]
@ef4 yeah, this is your call entirely. I'm not complaining. I do think releasing a major rev might be helpful, as apps pinned to the point releases currently break on update with a syntax error emitted from deep in the bowels.

I've definitely seen the full es6 transpiler be a build speed hit on apps before, and disabled it or recommended disabling it because it made for a poor development experience.

I have personal concerns about transpiler interop- if you test against transpiler A and transpiler B differs slightly in implementation.

I wish there was a way to tell ember-cli that your addon will require a certain process in the build pipeline. A similar example would be emblem templates- how you must bring your own compiler for them today. I would suggest that if you want es6 syntax internally you compile to es5+modules before a CLI app consumes you. And also suggest that ember-cli should not be passing addons through es6 transpilation by default.

tl;dr this is totally your choice, but you are definitely pushing the vanguard by including es6 sources :-) I would prefer to see 0.16.x be non-es6 safe and a 0.17 released that requires es6.

**Comment by ef4** [comment]
Yeah, definitely at a minimum shouldn't have changed the requirements in a minor release. I will do another minor release to fix, and if we start requiring ES2015 it will be in a major release.

**Comment by ef4** [comment]
Looks like https://github.com/ember-cli/ember-cli/pull/3166 will actually let an addon like this one use the 6to5 addon without pulling it into the rest of the app.
